pkg/config/expand.go

```diff
@@ -868,7 +868,11 @@ func expandSimpleVariable(context varContext, trackModuleGraph bool) (string, er
 		// ensure that the target module outputs the value in the root module
 		// state and not just internally within its deployment group
 		if !slices.ContainsFunc(toMod.Outputs, func(o modulereader.OutputInfo) bool { return o.Name == varRef.name }) {
-			toMod.Outputs = append(toMod.Outputs, modulereader.OutputInfo{Name: varRef.name})
+			toMod.Outputs = append(toMod.Outputs, modulereader.OutputInfo{
+				Name:        varRef.name,
+				Description: fmt.Sprintf("Automatically-generated output from module '%s'", toMod.ID),
+				Sensitive:   true,
```

Comment:
Does this happen after we add outputs requested by the user? So that if a user requested output and an IGC output align it will not be marked sensitive.

Reply:
The outputs specified by the user (which can be sensitive true/false) are not altered by the automatic addition of outputs. Additionally, the description will be set to the default value rather than the one for IGC references contained in this PR.